haskell/haddock.bzl

```diff
@@ -216,10 +221,11 @@ def _haskell_doc_rule_impl(ctx):
     if ctx.attr.index_transitive_deps:
         # Include all packages in the unified index.
         for package_id in html_dict_copied:
-            args.add("--read-interface=../{0},{1}".format(
-                package_id,
-                haddock_dict[package_id].path,
-            ))
+            if package_id in haddock_dict:
```

Comment:
That was needed to avoid an error for prebuild packages without haddock, but I've changed the way their `HaddockInfo` is generated so that the `transitive_html` and `transitive_haddocks` dicts have the same shape (though I'm not really fond of the fact that we manipulate two different dicts with the implicit invariant that they have the same keys, but that's out of scope for this PR)